orderbook.js

```diff
@@ -0,0 +1,41 @@
+const orderBook = (existingBook, incomingOrder) => {
+  // Check if exisitingBook has any values
+  if (existingBook.length <= 0) return [incomingOrder]
+
+  // Loop through exisiting book using a for..of loop. Also extracting iteration indexes
+  for (const [index, currentOrder] of existingBook.entries()) {
+    // Put special criteria into own variable, to clean up code
+    // eslint-disable-next-line max-len
+    const misMatchCriteria = ((incomingOrder.type === 'sell' && (incomingOrder.price <= currentOrder.price)) || (incomingOrder.type === 'buy' && incomingOrder.price >= currentOrder.price))
+    const matchingPriceCriteria = currentOrder.price === incomingOrder.price
+
+    if (currentOrder.type !== incomingOrder.type && misMatchCriteria) {
+      if (currentOrder.quantity === incomingOrder.quantity) {
+        existingBook.splice(index, 1)
+
+        return existingBook
+      } else if (currentOrder.quantity > incomingOrder.quantity && matchingPriceCriteria) {
```

Comment:
do you not want to process orders for the misMatchCriteria if current order quantity is greater? even though both seller and buyer benefit?

minor since its extra credit question